Make default view search case-insensitive

pspiller opened the pull request:
The search field in collection views is currently case-sensitive,
which means searching for e.g. 'shoe' won't return results containing
'Shoe'. This commit:

- Adds a new ILIKE query operator, which is the same as LIKE but
  converts both operands to lowercase before comparing.
- Makes the default search filter operator ILIKE rather than LIKE, so
  that searches are case-insensitive.

franzwilding commented on src/Bundle/CoreBundle/Service/GraphQLDoctrineFilterQueryBuilder.php:
```diff
@@ -144,6 +144,8 @@ private function getQueryBuilderComposite(array $filterInput)
                         case 'LIKE':
                             return $this->expr->like($leftSide, $rightSide);
+                        case 'ILIKE':
+                            return $this->expr->like($this->expr->lower($leftSide), $this->expr->lower($rightSide));
```
We need a test for this statement!

franzwilding commented:
Thank you, this is a very good PR, however we need an automatic test for the query builder change. Do you want to write one?

franzwilding commented:
@pspiller Can you write a test case for the new ILIKE operator?

pspiller commented:
No problem, I'd be happy to add some tests.

pspiller commented:
I've added some tests for the`ILIKE` operator. The tests in `testBuildingSimpleORandANDFilter` were duplicated and changed to use `ILIKE`, and I changed half the `LIKE` operators in `testBuildingComplexNestedFilter` to `ILIKE`.
